### ISSUE TITLE
GoCD doesn't parse a JUnit XML produced by  pytest

### ISSUE THREAD

**Issue by asavchkov** [gocd/gocd#11159]
##### Issue Type
- Bug Report

##### Summary
GoCD doesn;t process a valid JUnit XML result file

###### Basic environment details

* GoCD Version: 22.3.0 (15301
* JAVA Version: 17.0.5
* OS: Linux 5.15.0-56-generic

##### Steps to Reproduce

1. Configure a job with a pytest test with a --junitxml=test_results.xml option and a test  type artifact out of test_results.xml. This is the contents of my test_results.xml:
```
<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite errors="0" failures="0" hostname="57ceaf161752" name="pytest" skipped="0" tests="1" time="17.323" timestamp="2023-01-04T17:55:43.134227"><testcase classname="tests.auth_test.SimpleAuthTest" file="tests/auth_test.py" line="24" name="test_backup_via_unprivileged_user" time="15.839"></testcase></testsuite></testsuites>
```
2.  Run the job and check the Tests tab

##### Expected Results
The summary on the tab displays the number of tests run by the job (in my case - one test)

##### Actual Results
The summary on the tab displays:
```
Tests run: 0 , Failures: 0 , Not run: 0 , Time: .000 seconds.
No Tests Run
This project doesn't have any tests
```
There are no errors in the server log.

##### Any other info
The Artifacts tab shows the following folder structure
```
testoutput
  result
    index.html
    test_results.xml
```
suggesting that the test artifact is correctly set up and recognised by GoCD.

**Comment by chadlwilson** [comment]
Yeah, the problem here is the single-line XML pytest is producing.

A possible workaround for now is to look for a way to get pytest to produce non-minified XML (with the XML declaration on it's own line at the start). GoCD has some very old, very dumb logic to merge the JUnit report files together for reporting which is ignoring the entire line in that file since the line contains `<?xml`.

**Comment by asavchkov** [comment]
Many thanks for pointing out the single-line XML. After I added a newline after the `<?xml ... ?>` declaration the Tests tab started to show the correct test statistics. Looking forward to 23.1.0 to remove the workaround.

**Comment by chadlwilson** [comment]
It looks like this is a very old problem and an earlier attempt to fix it was reverted (see #266, #983, #1132, #4732, #5319 and probably others) due to performance/memory usage issues. Will see if my still-dumb-but-simple workaround is performant enough to not suffer from the same issues.

This particular feature is not particularly well loved or used (hence how ugly it looks in the current design!) so there's probably a limit to how much effort is worth putting in.
